src/devices_models/device_constructors/branch_constructor.jl

```diff
@@ -102,7 +102,12 @@ function construct_device!(
     if has_subnetworks(network_model)
         devices =
             get_available_components(T, sys, get_attribute(device_model, "filter_function"))
-        branch_rate_bounds!(container, devices, device_model, S)
+        branch_rate_bounds!(
+            container,
+            devices,
+            device_model,
+            typeof(network_model).parameters[1],
```

Comment:
This was an existing bug because `S` was not defined. Please confirm the change.

Reply:
This is correct. We probably should extend the tests to cover this area of the code